packages/next/src/client/components/app-router.tsx

```diff
@@ -329,7 +328,10 @@ function Router({
     // TODO-APP: Should we listen to navigateerror here to catch failed
     // navigations somehow? And should we call window.stop() if a SPA navigation
     // should interrupt an MPA one?
-    use(unresolvedThenable)
+    // NOTE: This is intentionally using `throw` instead of `use` because we're
+    // inside an externally mutable condition (pushRef.mpaNavigation), which
+    // violates the rules of hooks.
+    throw unresolvedThenable
```

Comment:
We'd have to refactor this when we get rid of the throw promise pattern in React, right?

Reply:
Yeah but it's not hard. For example, we could extract it into a separate component:

```js
return <SuspendIndefinitely>
```

Considered doing that in this PR but wanted to minimize the diff.